pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/measurements/tests/Test_TNCuda_Expval_dev.cpp

```diff
@@ -0,0 +1,430 @@
+
+//         mps_state.applyOperations({{"Hadamard"}, {"Hadamard"}, {"Hadamard"}},
+//                                   {{0}, {1}, {2}}, {{false}, {false},
+//                                   {false}});
+
+//         auto m = MeasurementsTNCuda<TensorNetT>(mps_state);
+
+//         auto X0 =
+//             std::make_shared<NamedObsT>("PauliX",
+//             std::vector<std::size_t>{0});
+//         auto Z1 =
+//             std::make_shared<NamedObsT>("PauliZ",
+//             std::vector<std::size_t>{1});
+
+//         auto ob = TensorProdObsT::create({X0, Z1});
+//         auto res = m.expval(*ob);
+//         CHECK(res == Approx(ZERO));
+//     }
+
+//     SECTION("Using HHH") {
+//         std::size_t bondDim = GENERATE(2);
+//         std::size_t num_qubits = 3;
+//         std::size_t maxBondDim = bondDim;
+
+//         TensorNetT mps_state{num_qubits, maxBondDim};
+
+//         mps_state.applyOperations({{"Hadamard"}, {"Hadamard"}, {"Hadamard"}},
+//                                   {{0}, {1}, {2}}, {{false}, {false},
+//                                   {false}});
+
+//         auto m = MeasurementsTNCuda<TensorNetT>(mps_state);
+
+//         auto H0 = std::make_shared<NamedObsT>("Hadamard",
+//                                               std::vector<std::size_t>{0});
+//         auto H1 = std::make_shared<NamedObsT>("Hadamard",
+//                                               std::vector<std::size_t>{1});
+//         auto H2 = std::make_shared<NamedObsT>("Hadamard",
+//                                               std::vector<std::size_t>{2});
+
+//         auto ob = TensorProdObsT::create({H0, H1, H2});
+//         auto res = m.expval(*ob);
+//         CHECK(res == Approx(INVSQRT2 / 2));
+//     }
+// }
+
+// TEMPLATE_TEST_CASE("Test expectation value of HamiltonianObs",
+//                    "[MPSTNCuda_Expval]", float, double) {
+//     using TensorNetT = MPSTNCuda<TestType>;
+//     using NamedObsT = NamedObsTNCuda<TensorNetT>;
+//     using HamiltonianObsT = HamiltonianTNCuda<TensorNetT>;
+//     auto ONE = TestType(1);
+//     SECTION("Using XZ") {
+//         std::size_t bondDim = GENERATE(2);
+//         std::size_t num_qubits = 3;
+//         std::size_t maxBondDim = bondDim;
+
+//         TensorNetT mps_state{num_qubits, maxBondDim};
+
+//         mps_state.applyOperations({{"Hadamard"}, {"Hadamard"}, {"Hadamard"}},
+//                                   {{0}, {1}, {2}}, {{false}, {false},
+//                                   {false}});
+
+//         auto m = MeasurementsTNCuda<TensorNetT>(mps_state);
+
+//         auto X0 =
+//             std::make_shared<NamedObsT>("PauliX",
+//             std::vector<std::size_t>{0});
+//         auto Z1 =
+//             std::make_shared<NamedObsT>("PauliZ",
+//             std::vector<std::size_t>{1});
+
+//         auto ob = HamiltonianObsT::create({TestType{1}, TestType{1}}, {X0,
+//         Z1}); auto res = m.expval(*ob); CHECK(res == Approx(ONE));
+//     }
+// }
```

Comment:
What is the idea for this part? Are we going to keep it commented until some condition is satisfied? Or are we going to remove it?

Reply:
I will remove the full file. it was just for developing :sweat_smile: